pkg/network/usm/ebpf_ssl.go

```diff
@@ -318,6 +317,7 @@ func (o *sslProgram) Start() {
 
 func (o *sslProgram) Stop() {
 	o.perfHandler.Stop()
+	o.watcher.Stop()
```

Comment:
o.watcher could be nil if we don't Start()

Reply:
the only scenario in which we do not call `.Start` is if we shouldn't capture tls traffic, in such a scenario we don't call .Stop method as well

Reply:
Adding the program to `subprogram` array:
https://github.com/DataDog/datadog-agent/blob/b87a657c4a89383e7659c8b800ec5b30746643aa/pkg/network/usm/ebpf_main.go#L157-L161


Calling start and stop:
https://github.com/DataDog/datadog-agent/blob/b87a657c4a89383e7659c8b800ec5b30746643aa/pkg/network/usm/ebpf_main.go#L255-L277

Reply:
it's true today, but what about tomorrow, the code path would change and may segv

Reply:
that's true, but regardless of my change we'll have to change many methods due to that, so I'm postponning that for the time we do (if we do)